### PR TITLE
chore: Fix build for clustering

### PR DESCRIPTION
**Pull request by lesam**
Closes #

Describe your proposed changes here.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Feature flagged (if modified API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: services/meta/data.go [hunk 1]
```diff
@@ -1363,12 +1363,15 @@ func (sgi ShardGroupInfo) clone() ShardGroupInfo {
...
 }
 
-// ShardFor returns the ShardInfo for a Point.
-func (sgi *ShardGroupInfo) ShardFor(p models.Point) ShardInfo {
+type hashIDer interface {
+	HashID() uint64
+}
+
+// ShardFor returns the ShardInfo for a Point or other hashIDer.
+func (sgi *ShardGroupInfo) ShardFor(p hashIDer) ShardInfo {
```
Comment (lesam) [inline services/meta/data.go:1371]: `hashIDer` is a much easier requirement than `Point` (an interface subset)

Comment (danxmoran) [reply services/meta/data.go:1371]: Wouldn't have thought of this, I like it!

**Review by danxmoran — APPROVED** [review]
LGTM, the ImportShard impl looks identical to the old code

File: services/meta/data.go (hunk 1 above)
Comment (dgnorton) [reply services/meta/data.go:1371]: Since `ShardFor` is public, `hashIDer` should probably be public also?

**Comment by lesam** [comment]
> LGTM, the ImportShard impl looks identical to the old code

Yeah I just reverted the PR that removed it with `git revert`

File: services/meta/data.go (hunk 1 above)
Comment (lesam) [reply services/meta/data.go:1371]: The interface only needs to be exported if we want to re-use it elsewhere, which we don't (I think), and also this package would not be the right one to export from.

I have validated locally that I can use this to fix the clustering build.

**Pull request merged**